libs/actions/sendmail.c

```diff
@@ -0,0 +1,226 @@
+/*
+	Copyright (C) 2015 CurlyMo & Niek
+
+	This file is part of pilight.
+
+	pilight is free software: you can redistribute it and/or modify it under the
+	terms of the GNU General Public License as published by the Free Software
+	Foundation, either version 3 of the License, or (at your option) any later
+	version.
+
+	pilight is distributed in the hope that it will be useful, but WITHOUT ANY
+	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with pilight. If not, see	<http://www.gnu.org/licenses/>
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "action.h"
+#include "log.h"
+#include "dso.h"
+#include "json.h"
+#include "common.h"
+#include "config.h"
+#include "pilight.h"
+#include "mail.h"
+#include "sendmail.h"
+#include "settings.h"
+
+
+//check arguments and settings
+static int actionSendmailArguments(struct JsonNode *arguments) {
+	struct JsonNode *jsubject = NULL;
+	struct JsonNode *jmessage = NULL;
+	struct JsonNode *jto = NULL;
+	struct JsonNode *jvalues = NULL;
+	struct JsonNode *jchild = NULL;
+	struct JsonNode *temp = NULL;
+	char *stmp = NULL;
+	int nrvalues = 0, itmp = 0;
+	
+	jsubject = json_find_member(arguments, "SUBJECT");
+	jmessage = json_find_member(arguments, "MESSAGE");
```

Comment:
Correct,  these arguments are optional and if omitted the corresponding 
setting will be used.



CurlyMoo schreef op 17-3-2015 om 19:01:
>
> In libs/actions/sendmail.c 
> <https://github.com/pilight/pilight/pull/155#discussion_r26598639>:
>
> > +#include "settings.h"
> > +
> > +
> > +//check arguments and settings
> > +static int actionSendmailArguments(struct JsonNode *arguments) {
> > +	struct JsonNode *jsubject = NULL;
> > +	struct JsonNode *jmessage = NULL;
> > +	struct JsonNode *jto = NULL;
> > +	struct JsonNode *jvalues = NULL;
> > +	struct JsonNode *jchild = NULL;
> > +	struct JsonNode *temp = NULL;
> > +	char *stmp = NULL;
> > +	int nrvalues = 0, itmp = 0;
> > +	
> > +	jsubject = json_find_member(arguments, "SUBJECT");
> > +	jmessage = json_find_member(arguments, "MESSAGE");
>
> I don't see you check for the |SUBJECT| and |TO| pressence?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/pilight/pilight/pull/155/files#r26598639>.
>